site/src/job_queue.rs

```diff
@@ -24,19 +27,118 @@ async fn enqueue_master_commits(ctxt: &Arc<SiteCtxt>) {
                 &master_commit.parent_sha,
                 pr,
                 master_commit.time,
-                BenchmarkRequestStatus::WaitingForParent,
+                BenchmarkRequestStatus::ArtifactsReady,
                 "",
                 "",
             );
             conn.insert_benchmark_request(&benchmark).await;
         }
     }
+    Ok(())
+}
+
+/// This function requires the sorts the priority order of the queue. It assumes
+/// that the only status is `ArtifactsReady`
+fn sort_benchmark_requests<'a>(
+    done: &mut HashSet<String>,
+    unordered_queue: &'a mut [BenchmarkRequest],
+) -> &'a [BenchmarkRequest] {
+    // A topological sort, where each "level" is additionally altered such that
+    // try commits come first, and then sorted by PR # (as a rough heuristic for
+    // earlier requests).
+
+    // Ensure the queue is ordered by status and the `commit_type`
+    unordered_queue.sort_unstable_by_key(|bmr| (bmr.status.rank(), bmr.commit_type.rank()));
```

Comment:
We assume that the status is "ArtifactsReady", so we don't need to sort by it, right? I don't think that we need to sort at all, because `partition_in_place` doesn't maintain relative order anyway (`The relative order of partitioned items is not maintained.` from https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.partition_in_place).

What we should do though is add an assert that all items in `unordered_queue` have status "ArtifactsReady".